### PR TITLE
Add home activity history view

### PR DESCRIPTION
**Pull request by Pierre-Gilles**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affect the code, did you write the tests? (8 new server tests: lib + controller)
- [x] Are tests passing? (`npm test` device suites: 261 passing; front has no unit-test script)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (both front/server, all files pass `prettier --check`)
- [x] If you are adding a new feature/service, did you run the integration comparator? (`npm run compare-translations` on front — en/fr/de complete)
- [x] Did you test this pull request in real life? With real devices? (tested end-to-end with a seeded local instance — screenshots below workflow; not yet with real devices)
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (apidoc comments added on the new endpoint)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation?
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend?

### Description of change

This PR adds a new **Activity** page (`/dashboard/history`, new "Activity" tab in the header) that shows a live, visual timeline of everything happening in the home: doors opening, motion detected, lights turned on/off, sensor values… It is designed to stay readable even for homes with 100–200 devices reporting states continuously.

**Front (the main part):**
- Visual timeline grouped by day ("Today", "Yesterday", then full dates), with a colored icon per event family, device name, human-readable action ("Opened", "Motion detected", "Turned on", "21.5 °C"…), room badge and time.
- **Burst grouping**: consecutive states of the same device feature are collapsed into a single line with a `×N` badge, expandable to see each timestamped occurrence — this is what keeps the feed readable with chatty sensors.
- Filter chips by event family (Openings, Motion & presence, Buttons & remotes, Lights & switches, Climate, Safety, Energy, Other), room selector and device search (server-side filtering, debounced).
- **Real-time**: new states arrive live through the existing websocket. When the user has scrolled down, events are not inserted (the content would move under their eyes): a floating "N new events" pill appears and scrolls back to top on click.
- Infinite scroll with cursor pagination (IntersectionObserver + "Load more" fallback), empty state, mobile responsive, en/fr/de translations, demo mode data.

**Server:**
- New endpoint `GET /api/v1/device_feature/states_history`: returns recent device states across all devices from DuckDB, most recent first, decorated with device feature / device / room metadata. Supports cursor pagination (`before` / `take`, capped at 500) and `categories` / `room_id` / `search` filters. The query is always constrained to existing device feature ids, so states of deleted features never consume rows of a page.

Screenshots (local instance with seeded data):
- Desktop timeline with live badge, filter chips and day sections
- "Openings" filter with ×N burst grouping and expandable occurrences
- Mobile view

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_012ZHMk4de6U51BGf6gnNqiR

---
_Generated by [Claude Code](https://claude.ai/code/session_012ZHMk4de6U51BGf6gnNqiR)_

## Summary by CodeRabbit

* **New Features**
  * Added an Activity/History timeline at `/dashboard/history` with day grouping (“today/yesterday”), room/category filtering, search, live “new events” buffering, and infinite loading.
  * Added an authenticated API endpoint to retrieve device feature state history with cursor pagination (including optional `before_id`).
  * Extended demo data with rooms and historical device feature state events.
* **Documentation**
  * Added/updated History page and navigation i18n strings across supported languages, including device value labels.
* **Tests**
  * Added backend/controller and library tests covering metadata, filtering, ordering, pagination, and edge cases.

### CODE REVIEW

File: front/src/routes/history/index.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        hasMore: newEvents.length === PAGE_SIZE,
+        loading: false,
+        initialized: true,
+        pendingLiveEvents: reset ? [] : prevState.pendingLiveEvents
+      }));
+    } catch (e) {
+      console.error(e);
+      this.setState({ loading: false, error: true, initialized: true });
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Pierre-Gilles) [reply front/src/routes/history/index.js:123]: Fixed in 5baab0a — the events array is now bounded to `MAX_EVENTS_IN_MEMORY` on the pagination path too.

I didn't apply the committable suggestion verbatim: it dropped the `else if (this.state.selectedDate)` branch (which powers the "jump to a past day" feature), and `.slice(0, MAX)` would keep the newest events while discarding the older ones just loaded — that stalls infinite scroll (the `before` cursor never advances) and drops what the user is scrolling toward. Instead I keep the trailing window with `events.slice(events.length - MAX_EVENTS_IN_MEMORY)`, dropping the already-scrolled-past newest events at the head.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: server/lib/device/device.getDeviceStatesHistory.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    SELECT device_feature_id, value, created_at
+    FROM t_device_feature_state
+    WHERE created_at < CAST(? AS TIMESTAMPTZ)
+    AND device_feature_id IN (${filteredFeatureIds.map(() => '?').join(',')})
+    ORDER BY created_at DESC
+    LIMIT ?
+  `;
+  const queryParams = [before.toISOString(), ...filteredFeatureIds, take];
+
+  const rows = await db.duckDbReadConnectionAllAsync(query, ...queryParams);
```
[the remark this answers is not in the document]
Comment (Pierre-Gilles) [reply server/lib/device/device.getDeviceStatesHistory.js:100]: Fixed in 5baab0a. `t_device_feature_state` has no unique row id (columns are `device_feature_id`, `value`, `created_at`), so I implemented keyset pagination on the compound key `(created_at, device_feature_id)`:

```sql
WHERE (created_at < ? OR (created_at = ? AND device_feature_id < ?))
ORDER BY created_at DESC, device_feature_id DESC
```

The endpoint now returns `device_feature.id` and accepts a `before_id` cursor that the front sends alongside `before`. States sharing the same `created_at` are no longer skipped across page boundaries. Added a regression test (`should not skip states sharing the same created_at when paginating`).

---
_Generated by [Claude Code](https://claude.ai/code)_
